engine/repositories/processor_checkout.go

```diff
@@ -64,6 +67,43 @@ func (s *Service) processCheckout(ctx context.Context, op *sdk.Operation) error
 		}
 	}
 
+	if op.Setup.Checkout.CheckSignature && op.Setup.Checkout.Commit != "" {
+		log.Debug(ctx, "retrieve gpg key id")
+		c, err := gitRepo.GetCommit(ctx, op.Setup.Checkout.Commit)
+		if err != nil {
+			return sdk.WithStack(err)
+		}
+
+		if c.GPGKeyID == "" {
+			op.Setup.Checkout.Result.CommitVerified = false
+			op.Setup.Checkout.Result.Msg = "commit not signed"
+			return nil
+		}
+		ctx = context.WithValue(ctx, cdslog.GpgKey, c.GPGKeyID)
+		op.Setup.Checkout.Result.SignKeyID = c.GPGKeyID
+
+		// Retrieve gpg public key
+		key, err := s.Client.UserGpgKeyGet(ctx, c.GPGKeyID)
+		if err != nil {
+			op.Setup.Checkout.Result.CommitVerified = false
+			op.Setup.Checkout.Result.Msg = fmt.Sprintf("commit signed but key %s not found in CDS: %v", c.GPGKeyID, err)
+			return nil
+		}
+
+		// Import gpg public key
+		fileName, _, err := sdk.ImportGPGKey(os.TempDir(), c.GPGKeyID, key.PublicKey)
+		if err != nil {
+			return err
+		}
+		log.Debug(ctx, "key: %s, fileName: %s imported", c.GPGKeyID, fileName)
+
+		// Check commit signature
+		if err := gitRepo.VerifyCommit(ctx, op.Setup.Checkout.Commit); err != nil {
+			return sdk.NewErrorFrom(sdk.ErrUnauthorized, "unable to verify commit signature")
```

Comment:
at line 88, we are not able to verify the commit.
as a resut we set `op.Setup.Checkout.Result` and we return nil

here we don't set `op.Setup.Checkout.Result` and we return an error.

shouldn't we behave the same ?

Reply:
fixed